typescript/infra/src/govern/HyperlaneHaasGovernor.ts

```diff
@@ -71,8 +77,52 @@ export class HyperlaneHaasGovernor extends HyperlaneAppGovernor<
   }
 
   async check(chainsToCheck?: ChainName[]) {
-    await this.icaChecker.check(chainsToCheck);
-    await this.coreChecker.check(chainsToCheck);
+    // Get all EVM chains from core config
+    const evmChains = this.coreChecker.getEvmChains();
+
+    // Mark any EVM chains that are not deployed
+    const appChains = this.coreChecker.app.chains();
+    for (const chain of evmChains) {
+      if (!appChains.includes(chain)) {
+        this.coreChecker.addViolation({
+          type: ViolationType.NotDeployed,
+          chain,
+          expected: '',
+          actual: '',
+        });
+      }
+    }
+
+    // Finally, check the chains that were explicitly requested
+    // If no chains were requested, check all app chains
+    const chains =
+      !chainsToCheck || chainsToCheck.length === 0 ? appChains : chainsToCheck;
+    const failedChains: ChainName[] = [];
+    if (chainsToSkip.length > 0) {
+      rootLogger.info(
+        chalk.yellow('Skipping chains:', chainsToSkip.join(', ')),
+      );
+    }
+    await Promise.allSettled(
```

Comment:
fwiw we can use the output of this to get the success / errors, but do you prefer the try/catch and failedChains mutation in the map because you log the errors earlier or something?